### PR TITLE
[flutter_tools] add --uninstall-first flag and pipe it through to ios-deploy

### PR DESCRIPTION
**Pull request by christopherfujino**
Hopefully address https://github.com/flutter/flutter/issues/95582.

I ran the devicelab test locally and it removed the gallery app from my phone before installing it again.

### CODE REVIEW

File: dev/devicelab/lib/tasks/hot_mode_tests.dart [hunk 1]
```diff
@@ -33,7 +33,15 @@ TaskFunction createHotModeTest({String? deviceIdOverride, Map<String, String>? e
...
-      '--hot', '-d', deviceIdOverride!, '--benchmark', '--resident',  '--no-android-gradle-daemon', '--no-publish-port', '--verbose',
+      '--hot',
+      '-d',
+      deviceIdOverride!,
+      '--benchmark',
+      '--resident',
+      '--no-android-gradle-daemon',
+      '--no-publish-port',
+      '--verbose',
+      '--uninstall-first',
```
Comment (christopherfujino) [inline dev/devicelab/lib/tasks/hot_mode_tests.dart:44]: this is the only non-formatting change here

**Review by jmagman — APPROVED** [review]
LGTM with help text nit.
By the way it should be pretty trivial to get this working on Android if you pass `uninstallFirst` to `installApp`, then copy this logic to the top if it's true.
https://github.com/flutter/flutter/blob/1bb9eed4d0ecaf62cfe38e9760f71324b090aa02/packages/flutter_tools/lib/src/android/android_device.dart#L419-L420

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Review by jmagman — APPROVED** [review]

**Comment by christopherfujino** [comment]
Looks like this didn't work: https://ci.chromium.org/ui/p/flutter/builders/staging/Mac_ios%20hot_mode_dev_cycle_ios__benchmark/1342/overview

This is the latest build, 2 days after this PR landed, and I can see that it is successfully uninstalling the app before running, and yet `flutter doctor -v` still shows the debug symbol error, and the observatory did not connect after 30 seconds (although it never fully timed out)

**Pull request merged**